src/consumer/component/consumer-component.js

```diff
@@ -411,7 +411,8 @@ export default class Component {
     if (this.files) await this.files.forEach(file => file.write(undefined, force));
     if (this.dists && this._writeDistsFiles) await this.writeDists(consumer, false);
     if (withBitJson) await this.writeBitJson(bitDir, force);
-    if (withPackageJson) {
+    // make sure the project's package.json is not overridden by Bit
+    if (withPackageJson && bitDir !== consumer.getPath()) {
```

Comment:
I would consider adding this check to the function that actually writes the file, which is in consumer/component/package-json.js write().

Reply:
seems like many tests are now failing. @GiladShoham is working on it